<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>preferences</class>
 <widget class="QWidget" name="preferences">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>702</width>
    <height>401</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>702</width>
    <height>401</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>3333333</width>
    <height>3333333</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <property name="windowIcon">
   <iconset theme="settings">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QWidget" name="widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>128</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>256</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="baseSize">
       <size>
        <width>160</width>
        <height>0</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLineEdit" name="lineEdit_3">
         <property name="placeholderText">
          <string>Search</string>
         </property>
         <property name="clearButtonEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTreeWidget" name="treeWidget">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <attribute name="headerVisible">
          <bool>false</bool>
         </attribute>
         <column>
          <property name="text">
           <string>1</string>
          </property>
         </column>
         <item>
          <property name="text">
           <string>General</string>
          </property>
          <property name="icon">
           <iconset theme="computer">
            <normaloff>.</normaloff>.</iconset>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEnabled</set>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Arduino</string>
          </property>
          <property name="icon">
           <iconset theme="cpu">
            <normaloff>.</normaloff>.</iconset>
          </property>
          <property name="flags">
           <set>ItemIsSelectable|ItemIsEnabled</set>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QStackedWidget" name="stacked_widget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>512</width>
        <height>0</height>
       </size>
      </property>
      <property name="baseSize">
       <size>
        <width>512</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="general_preferences">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QFrame" name="telemetry_frame">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Telemetry server</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignHCenter|Qt::AlignTop</set>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QFormLayout" name="formLayout_2">
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
             </property>
             <property name="rowWrapPolicy">
              <enum>QFormLayout::WrapLongRows</enum>
             </property>
             <property name="labelAlignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
             <property name="formAlignment">
              <set>Qt::AlignRight|Qt::AlignTop|Qt::AlignTrailing</set>
             </property>
             <item row="2" column="0">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Address:</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Port:</string>
               </property>
               <property name="buddy">
                <cstring>udp_port</cstring>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QSpinBox" name="udp_port">
               <property name="minimum">
                <number>10000</number>
               </property>
               <property name="maximum">
                <number>32668</number>
               </property>
               <property name="value">
                <number>20777</number>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="udp_host">
               <property name="placeholderText">
                <string>127.0.0.1</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="behaviour_frame">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QLabel" name="label_8">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>16</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>16</height>
              </size>
             </property>
             <property name="baseSize">
              <size>
               <width>0</width>
               <height>16</height>
              </size>
             </property>
             <property name="text">
              <string>Application behaviour</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignHCenter|Qt::AlignTop</set>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QFormLayout" name="formLayout_7">
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
             </property>
             <property name="rowWrapPolicy">
              <enum>QFormLayout::WrapLongRows</enum>
             </property>
             <property name="labelAlignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
             <property name="formAlignment">
              <set>Qt::AlignRight|Qt::AlignTop|Qt::AlignTrailing</set>
             </property>
             <item row="1" column="0">
              <widget class="QLabel" name="label_7">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Preferred ga&amp;me:</string>
               </property>
               <property name="buddy">
                <cstring>gamesComboBox</cstring>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QComboBox" name="gamesComboBox"/>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="startServiceAtApplicationLaunchLabel">
               <property name="text">
                <string>S&amp;tart service at application launch</string>
               </property>
               <property name="wordWrap">
                <bool>false</bool>
               </property>
               <property name="buddy">
                <cstring>autostart</cstring>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QCheckBox" name="autostart"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="arduino_preferences">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>320</width>
         <height>0</height>
        </size>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QFrame" name="frame">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>320</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QLabel" name="label_9">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>16</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>16</height>
              </size>
             </property>
             <property name="text">
              <string>Arduino</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QFormLayout" name="formLayout_8">
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
             </property>
             <property name="labelAlignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
             <property name="formAlignment">
              <set>Qt::AlignRight|Qt::AlignTop|Qt::AlignTrailing</set>
             </property>
             <item row="1" column="0">
              <widget class="QLabel" name="label_10">
               <property name="text">
                <string>COM port:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_11">
               <property name="text">
                <string>&amp;Baud rate:</string>
               </property>
               <property name="buddy">
                <cstring>arduino_baud_rate</cstring>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBox" name="arduino_baud_rate">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="minimum">
                <number>300</number>
               </property>
               <property name="maximum">
                <number>250000</number>
               </property>
               <property name="singleStep">
                <number>100</number>
               </property>
               <property name="value">
                <number>115200</number>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="autoStartArduinoLabel">
               <property name="text">
                <string>Start Arduino serialCommunication with &amp;telemetry server</string>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
               <property name="buddy">
                <cstring>arduino_autostart</cstring>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QCheckBox" name="arduino_autostart"/>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="arduino_com_port">
               <property name="placeholderText">
                <string>/dev/ttyACM0</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>treeWidget</sender>
   <signal>itemSelectionChanged()</signal>
   <receiver>preferences</receiver>
   <slot>change_preferences_panel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>udp_host</sender>
   <signal>textChanged(QString)</signal>
   <receiver>preferences</receiver>
   <slot>on_preferences_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>463</x>
     <y>64</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>udp_port</sender>
   <signal>valueChanged(int)</signal>
   <receiver>preferences</receiver>
   <slot>on_preferences_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>463</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gamesComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>preferences</receiver>
   <slot>on_preferences_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>524</x>
     <y>249</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autostart</sender>
   <signal>stateChanged(int)</signal>
   <receiver>preferences</receiver>
   <slot>on_preferences_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>524</x>
     <y>271</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>arduino_com_port</sender>
   <signal>textChanged(QString)</signal>
   <receiver>preferences</receiver>
   <slot>on_preferences_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>arduino_baud_rate</sender>
   <signal>valueChanged(QString)</signal>
   <receiver>preferences</receiver>
   <slot>on_preferences_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>arduino_autostart</sender>
   <signal>stateChanged(int)</signal>
   <receiver>preferences</receiver>
   <slot>on_preferences_updated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>change_preferences_panel()</slot>
  <slot>on_preferences_updated()</slot>
 </slots>
</ui>
